src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyTest.java

```diff
@@ -135,4 +135,34 @@ public void discoverNodesOneAddressOnePort()
         assertEquals(new Address(publicAddress, port), node.getPublicAddress());
         assertFalse(iterator.hasNext());
     }
+
+    @Test
+    public void validateValidRegion() {
+        List<String> awsRegions = Arrays.asList(
+            "us-east-2", "us-west-2", "ap-south-1", "ap-northeast-3", "ap-southeast-2",
+            "ca-central-1", "cn-north-1", "cn-northwest-1", "us-gov-east-1");
+
+        awsRegions.forEach(region -> {
+            AwsDiscoveryStrategy awsDiscoveryStrategy = spy(new AwsDiscoveryStrategy(Collections.emptyMap(), null, mockClient));
```

Comment:
Why do you need to `spy` it here? Wouldn't it work if you remove these two lines?

Reply:
It depends. If I want it to work without the spy, I'll have to pass a Map to the constructor with the region property. Else, I'll have to mock the return for `awsDiscoveryStrategy.getCurrentRegion()`.

I don't mind either of the solutions, but I'll go with the Map one to avoid the spy and use the actual class and method call.

Reply:
Yes, I kept the old tests after changing the method to package-private. I'll clean it up.